stellargraph/core/validation.py

```diff
@@ -40,3 +41,28 @@ def require_dataframe_has_columns(name, df, columns):
         raise ValueError(
             f"{name}: expected {comma_sep(columns)} columns, found: {comma_sep(df.columns)}"
         )
+
+
+def require_integer_in_range(x, variable_name, min_val=-np.inf, max_val=np.inf):
+    """
+    A function to verify that a variable is an integer in a specified closed range.
+    Args:
+        x: the variable to check
+        variable_name (str): the name of the variable to print out in error messages
+        min_val: the minimum value that `x` can attain
+        min_val: the maximum value that `x` can attain
+    """
+    if min_val == -np.inf:
```

Comment:
It would be very slightly better for this to be inside the `if x < min_val or ...:` branch, so that we don't waste time doing these checks or string formatting if `x` is valid.